include/internal/list.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_INTERNAL_LIST_H
+# define OSSL_INTERNAL_LIST_H
+# pragma once
+
+# include <string.h>
+
+/* Define a list structure */
+# define OSSL_LIST(name) OSSL_LIST_ ## name
+
+/* Define fields to include an element of a list */
+# define OSSL_LIST_MEMBER(name, type)                                       \
+    struct {                                                                \
+        type *next, *prev;                                                  \
```

Comment:
Shouldn't this be `car` and `cdr`? :wink:

Reply:
I thought about naming them such, however I think there might be objections.

Reply:
Yeah...  among others, it's not exactly right, like I said.  `car` is supposed to point at the value of a cons, it looks weird if it's used as `prev`.

So we can save this one for the single linked list types :wink: